### PR TITLE
New inline keyboard example

### PR DESCRIPTION
**Pull request by sahilsharma811**
This is an example of a bot with an inline keyboard that can handle user's response with a dedicated callback query handler. The bot implements a Conversation Handlers to handle the response of the user, which is sent by the inline keyboard.

### CODE REVIEW

File: examples/inlinekeyboard2.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        chat_id=query.message.chat_id,
+        message_id=query.message.message_id,
+        text=u"Second CallbackQueryHandler"
+    )
+    return
+
+def error(bot, update, error):
+    logger.warn('Update "%s" caused error "%s"' % (update, error))    
+
+def  main():
```
Comment (bomjacob) [inline examples/inlinekeyboard2.py]: Extra space

File: examples/inlinekeyboard2.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        
+    # Create the Updater and pass it your bot's token. Below is an example token
+    updater = Updater("476482598:AAGR26TlclQP1ZdsAbBuGy9AeCkybP8k-kA")
```
Comment (bomjacob) [inline examples/inlinekeyboard2.py]: Please remove the token ;)

File: examples/inlinekeyboard2.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+    return SECOND
+
+def second(bot, update):
+    query = update.callback_query
+    bot.edit_message_text(
+        chat_id=query.message.chat_id,
+        message_id=query.message.message_id,
+        text=u"Second CallbackQueryHandler"
+    )
+    return
```
Comment (bomjacob) [inline examples/inlinekeyboard2.py]: Unnecessary return statement

File: examples/inlinekeyboard2.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+
+def start(bot, update):
+    user = update.message.from_user
+    logger.info("User %s started the conversation." % user.first_name)
+    keyboard = [
+        [InlineKeyboardButton(u"Next", callback_data=str(FIRST))]
+    ]
+    reply_markup = InlineKeyboardMarkup(keyboard)
+    update.message.reply_text(
+        u"Start handler, Press next",
```
Comment (bomjacob) [inline examples/inlinekeyboard2.py]: Do we mark strings in the other examples as unicode?

File: examples/inlinekeyboard2.py [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+logging.basicConfig(format='%(asctime)s - %(name)s - %(levelname)s - %(message)s',
+                    level=logging.INFO)
+
+logger = logging.getLogger(__name__)
+
+FIRST, SECOND = range(2)
+
+def start(bot, update):
+    user = update.message.from_user
+    logger.info("User %s started the conversation." % user.first_name)
```
Comment (bomjacob) [inline examples/inlinekeyboard2.py]: Should look like `logger.info("User %s started the conversation.", user.first_name)` for performence reasons (really just to teach users to do it properly)

File: examples/inlinekeyboard2.py (hunk 4 above)
Comment (sahilsharma811) [reply examples/inlinekeyboard2.py]: No! We are not marking strings in other example as unicode, but I think it's a good practice to mark strings as unicode.

File: examples/inlinekeyboard2.py (hunk 3 above)
Comment (sahilsharma811) [reply examples/inlinekeyboard2.py]: Removed

File: examples/inlinekeyboard2.py (hunk 2 above)
Comment (sahilsharma811) [reply examples/inlinekeyboard2.py]: Removed

File: examples/inlinekeyboard2.py (hunk 1 above)
Comment (sahilsharma811) [reply examples/inlinekeyboard2.py]: Removed

File: examples/inlinekeyboard2.py (hunk 5 above)
Comment (sahilsharma811) [reply examples/inlinekeyboard2.py]: Done!

File: examples/inlinekeyboard2.py (hunk 4 above)
Comment (sahilsharma811) [reply examples/inlinekeyboard2.py]: Should I keep it or make this example symmetrical with other examples?

**Comment by sahilsharma811** [comment]
@bomjacob. I've made all the changes as requested.

**Comment by vr2xiq** [comment]
is it possible to return different routes?
let's say 2 buttons inlinekeyboard
press 1 return third route
press 2 return forth route

**Comment by sahilsharma811** [comment]
It might be possible, but I'm not sure yet.

**Comment by Eldinnie** [comment]
It looks okay to me. But I have some questions/additions I would like to see.
First of all, the conversation is never ended. I would prefer if you return `ConversationHandler.END` at the end of `second`.

Secondly I do agree with @vr2xiq that it would be nice if the example would show a keyboard with a choice. So for example on start you have a choice of two buttons (`1` and `2`)
When selected `1` you will have a new choice of `3` and `4`, and when selected `4` you will have a second choice of `4` and `2` (just an example)
This will have a more elaborate view on `CallbackQqueryHandler`s in a `ConversationHandler` and will have to use the `pattern` argument to discern between them.

You think it's possible for you to adjust the example that way?

**Comment by sahilsharma811** [comment]
@Eldinnie I have added the ConversationHandler.END at the end of second.
For the keyboard which provide two or more choices, I will work on that ASAP. Till then we can merge this example so other developers can use the example to build awesome bots. Thanks

**Comment by tsnoam** [comment]
@bomjacob @Eldinnie 
I see there are commits following your review. Can you please check if we can merge it?

**Pull request closed without merging**